src/java/voldemort/client/ClientConfig.java

```diff
@@ -571,6 +577,52 @@ public ClientConfig setSocketTimeout(int socketTimeout, TimeUnit unit) {
         return this;
     }
 
+    public long getIdleConnectionTimeout(TimeUnit unit) {
+        if (idleConnectionTimeoutMs <= 0) {
+            return idleConnectionTimeoutMs;
+        }
+        return unit.convert(idleConnectionTimeoutMs, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Set the timeout for idle connections. Voldemort client caches all
+     * connections to the Voldemort server. This setting allows the a connection
+     * to be dropped, if it is idle for more than this time.
+     * 
+     * This could be useful in the following cases 1) Voldemort client is not
+     * directly connected to the server and is connected via proxy of firewall.
```

Comment:
Just to avoid confusion ...
"proxy of firewall"
should be ...
"a proxy or firewall"